pubsub/gcloud/aio/pubsub/subscriber_client.py

```diff
@@ -132,60 +162,56 @@ def create_subscription(self,
-
-        def run_forever(self, sub_keepalive: StreamingPullFuture) -> None:
+            url = f'{API_ROOT}/v1/{subscription}:pull'
+            headers = await self._headers()
+            body = {
+                'maxMessages': max_messages,
+            }
+            encoded = json.dumps(body).encode()
+            s = session or self.session
+            resp = await s.post(url, data=encoded, headers=headers)
+            resp = await resp.json()
+            messages = []
+            for m in resp['receivedMessages']:
+                messages.append(SubscriberMessage.from_api_dict(m))
+            return messages
+
+        async def acknowledge(self, subscription: str, ack_ids: List[str],
+                              *, session: Optional[Session] = None) -> None:
             """
-            Start the asyncio loop, running until it is either SIGTERM-ed or
-            killed by keyboard interrupt. The StreamingPullFuture parameter is
-            used to cancel subscription in the case that an unexpected
-            exception is thrown. You can also directly pass the `.subscribe()`
-            method call instead like so:
-                sub.run_forever(sub.subscribe(callback))
+            Acknowledge messages by ackIds
             """
-            try:
-                self.loop.run_forever()
-            except (KeyboardInterrupt, concurrent.futures.CancelledError):
-                pass
-            finally:
-                # 1. stop the `SubscriberClient` future, which will prevent
-                #    more tasks from being leased
-                if not sub_keepalive.cancelled():
-                    sub_keepalive.cancel()
-                # 2. cancel the tasks we already have, which should just be
-                #    `worker` instances; note they have
-                #    `except CancelledError: pass`
-                for task in asyncio.Task.all_tasks(loop=self.loop):
-                    task.cancel()
-                # 3. stop the `asyncio` event loop
-                self.loop.stop()
-
-        def _wrap_callback(self,
-                           callback: Callable[[SubscriberMessage], None]
-                           ) -> Callable[[Message], None]:
-            """Schedule callback to be called from the event loop"""
-            def _callback_wrapper(message: Message) -> None:
-                asyncio.run_coroutine_threadsafe(
-                    callback(  # type: ignore
-                        SubscriberMessage.from_google_cloud(
-                            message)),
-                    self.loop)
-
-            return _callback_wrapper
+            url = f'{API_ROOT}/v1/{subscription}:acknowledge'
+            headers = await self._headers()
+            body = {
+                'ackIds': ack_ids,
+            }
+            encoded = json.dumps(body).encode()
+            s = session or self.session
+            await s.post(url, data=encoded, headers=headers)
+            return
```

Comment:
nit: unnecessary `return` here, Python will already return `None` when you hit the end of a method.